test/wt_hang_analyzer/wt_hang_analyzer.py

```diff
@@ -0,0 +1,608 @@
+            "set print thread-events off",  # Suppress GDB messages of threads starting/finishing.
+            "file %s" % process_name,
+            "attach %d" % pid,
+            "info sharedlibrary",
+            "info threads",  # Dump a simple list of commands to get the thread name
+            "set python print-stack full",
+            # Lock the scheduler, before running commands, which execute code in the attached process.
+            "set scheduler-locking on",
+            dump_command,
+            "set confirm off",
+            "quit",
+        ]
+
+        call([dbg, "--quiet", "--nx"] +
+             list(itertools.chain.from_iterable([['-ex', b] for b in cmds])), logger)
+
+        root_logger.info("Done analyzing %s process with PID %d", process_name, pid)
+
+    @staticmethod
+    def get_dump_ext():
+        """Return the dump file extension."""
+        return "core"
+
+    @staticmethod
+    def _find_gcore():
+        """Find the installed gcore."""
+        dbg = "/usr/bin/gcore"
+        if os.path.exists(dbg):
+            return dbg
+
+        return None
+
+class LinuxProcessList(object):
+    """LinuxProcessList class."""
+
+    @staticmethod
+    def __find_ps():
+        """Find ps."""
+        return find_program('ps', ['/bin', '/usr/bin'])
+
+    def dump_processes(self, logger):
+        """Get list of [Pid, Process Name]."""
+        ps = self.__find_ps()
+
+        logger.info("Getting list of processes using %s", ps)
+
+        call([ps, "--version"], logger)
+
+        ret = callo([ps, "-eo", "pid,args"], logger)
+
+        buff = TextIOWrapper(BytesIO(ret))
+        csv_reader = csv.reader(buff, delimiter=' ', quoting=csv.QUOTE_NONE, skipinitialspace=True)
+
+        return [[int(row[0]), os.path.split(row[1])[1]] for row in csv_reader if row[0] != "PID"]
+
+def get_hang_analyzers():
+    """Return hang analyzers."""
+
+    dbg = None
+    ps = None
+    if sys.platform.startswith("linux"):
+        dbg = GDBDumper()
+        ps = LinuxProcessList()
+    elif _IS_WINDOWS or sys.platform == "cygwin":
+        dbg = WindowsDumper()
+        ps = WindowsProcessList()
+    elif sys.platform == "darwin":
+        dbg = LLDBDumper()
+        ps = DarwinProcessList()
+
+    return [ps, dbg]
+
+def check_dump_quota(quota, ext):
+    """Check if sum of the files with ext is within the specified quota in megabytes."""
+
+    files = glob.glob("*." + ext)
+
+    size_sum = 0
+    for file_name in files:
+        size_sum += os.path.getsize(file_name)
+
+    return size_sum <= quota
+
+def pname_match(exact_match, pname, processes):
+    """Return True if the pname matches in processes."""
+    pname = os.path.splitext(pname)[0]
+    for ip in processes:
+        if exact_match and pname == ip or not exact_match and ip in pname:
+            return True
+    return False
+
+# Basic procedure
+#
+# 1. Get a list of interesting processes
+# 2. Dump useful information or take dumps
```

Comment:
Minor nit: End the sentences with a period.

Reply:
Addressed in ef39187